### PR TITLE
feat: add pagination support to all list endpoints MAPCO-9304

### PR DESCRIPTION
**Pull request by ronenkapelian**
Adds `page` / `page_size` query parameters to all 5 list endpoints. Response shape changes from a raw array to `{ total, items }`.

## Endpoints affected
- `GET /v1/jobs`
- `GET /v1/stages`
- `GET /v1/jobs/:jobId/stages`
- `GET /v1/tasks`
- `GET /v1/stages/:stageId/tasks`

## Changes
- **`openapi_v1.yaml`** — source of truth updated with pagination params and paginated response schemas; `openapi3.yaml` and `src/openapi.d.ts` regenerated
- **`src/common/utils/pagination.ts`** — shared `paginationParamsToTakeAndSkip()` utility
- **Managers** — all 3 managers run `findMany` + `count` in parallel and return `{ total, items }`
- **`eslint.config.mjs`** — allowlist for `page_size` snake_case destructuring
- **Tests** — all existing tests updated; 15 new pagination edge-case tests added (page slicing, beyond-last-page)

## Breaking change
`T[]` → `{ total: number; items: T[] }` on all 5 endpoints.

**Consumer migration:**
- `response` → `response.items`
- `response.length` → `response.total`

Release as `0.3.0`.

### CODE REVIEW

File: src/common/utils/pagination.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const DEFAULT_PAGE = 1;
+export const DEFAULT_PAGE_SIZE = 10;
+
+/**
+ * Converts 1-based page / pageSize parameters into Prisma's take / skip values.
+ * @param page - 1-based page number (defaults to 1)
+ * @param pageSize - Number of items per page (defaults to DEFAULT_PAGE_SIZE)
+ * @returns Object containing `take` and `skip` values for Prisma queries
+ */
+export function paginationParamsToTakeAndSkip(page: number = DEFAULT_PAGE, pageSize: number = DEFAULT_PAGE_SIZE): { take: number; skip: number } {
```
Comment (CptSchnitz) [inline src/common/utils/pagination.ts]: create a pagination type

File: src/jobs/models/manager.ts [hunk 2]
```diff
@@ -27,26 +28,30 @@ export class JobManager {
...
+    const queryOptions = {
+      where,
+      include: { stage: filterParams.should_return_stages ?? false },
+      take,
+      skip,
+      orderBy: { creationTime: 'asc' as const },
+    };
 
-    const jobs = await this.prisma.job.findMany(queryBody);
+    const [jobs, total] = await Promise.all([this.prisma.job.findMany(queryOptions), this.prisma.job.count({ where })]);
```
Comment (CptSchnitz) [inline src/jobs/models/manager.ts]: consider if its better to use Promise.all or prisma.$transaction([]) api.
Promise.all is faster, but the count/jobs might not match.

File: src/stages/models/manager.ts [hunk 3]
```diff
@@ -169,7 +178,7 @@ export class StageManager {
   }
 
   @withSpanAsyncV4
-  public async getStagesByJobId(jobId: string, includeTasks?: boolean): Promise<StageModel[]> {
+  public async getStagesByJobId(jobId: string, query: StagesByJobIdQuery): Promise<StagesPaginatedResponse> {
```
Comment (CptSchnitz) [inline src/stages/models/manager.ts]: I dont think this endpoint needs pagination. it can be removed

File: src/jobs/models/manager.ts (hunk 2 above)
Comment (CptSchnitz) [inline src/jobs/models/manager.ts]: try creating a generic function to wrap all those pagination queries that does the count for you.

I gave claude a chance, you should check if it works:
```typescript
export async function paginate<Delegate extends FindManyDelegate, Args extends Prisma.Args<Delegate, 'findMany'>>(
  delegate: Delegate,
  args: Args,
  page: number = DEFAULT_PAGE,
  pageSize: number = DEFAULT_PAGE_SIZE
): Promise<PaginatedResult<Prisma.Result<Delegate, Args, 'findMany'>>> {
  const skip = (page - 1) * pageSize;
  const findMany = delegate.findMany as (a: unknown) => Promise<Prisma.Result<Delegate, Args, 'findMany'>>;
  const count = delegate.count as (a: unknown) => Promise<number>;

  const [items, total] = await Promise.all([findMany({ ...args, take: pageSize, skip }), count({ where: (args as { where?: unknown }).where })]);
  return { total, items };
}
```

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/stages/models/manager.ts (hunk 3 above)
Comment (ronenkapelian) [reply src/stages/models/manager.ts]: removed

File: src/jobs/models/manager.ts (hunk 2 above)
Comment (ronenkapelian) [reply src/jobs/models/manager.ts]: Used Promise.all on purpose — it's a list endpoint, so a rare off-by-one between total and items if a job gets created/deleted mid-request isn't worth worrying about, it'll be correct on the next fetch anyway. $transaction wouldn't even fix that: on Postgres's default isolation the two queries still don't share a snapshot, so we'd need RepeatableRead to actually guarantee consistency, and that kills the parallelism we're trying to get here. Not worth the tradeoff for a count.

File: src/common/utils/pagination.ts (hunk 1 above)
Comment (ronenkapelian) [reply src/common/utils/pagination.ts]: done

File: src/jobs/models/manager.ts (hunk 2 above)
Comment (ronenkapelian) [reply src/jobs/models/manager.ts]: done

File: src/common/utils/pagination.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  items: T;
+}
+
+/**
+ * Runs a paginated `findMany` and a matching `count` in parallel and returns `{ total, items }`.
+ *
+ * `items` is typed to the exact payload of `args` (relations from `include`/`select` are reflected),
+ * and `total` is the full number of rows matching `args.where`, independent of `page` / `pageSize`.
+ * Callers pass a Prisma model delegate (e.g. `prisma.job`) plus the usual `findMany` args minus
+ * `take` / `skip`, which are derived here from the 1-based `page` / `pageSize`.
```
Comment (CptSchnitz) [inline src/common/utils/pagination.ts]: this description is too much

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/common/utils/pagination.ts (hunk 4 above)
Comment (ronenkapelian) [reply src/common/utils/pagination.ts]: done

**Review by CptSchnitz — APPROVED** [review]
